### PR TITLE
Implementation of cli command for deleting users from an organization

### PR DESCRIPTION
**Pull request by rocknes**

### CODE REVIEW

File: vcd_cli/user.py [hunk 1]
```diff
@@ -143,3 +148,34 @@ def create(ctx, user_name, password, role_name, full_name, description, email,
...
+    Examples
+        vcd user delete 'my user'
+           deletes user with username "my user" from the current organization.
+
+    """  # NOQA
+    try:
+        client = ctx.obj['client']
+        org_name = ctx.obj['profiles'].get('org')
+        in_use_org_href = ctx.obj['profiles'].get('org_href')
+        org = Org(client, in_use_org_href, org_name == 'System')
```
Comment (sahithi) [inline vcd_cli/user.py]: Consider using is_sysAdmin() utils method, which is more descriptive to the code readers.

File: vcd_cli/user.py [hunk 2]
```diff
@@ -143,3 +148,34 @@ def create(ctx, user_name, password, role_name, full_name, description, email,
...
+              is_flag=True,
+              callback=abort_if_false,
+              expose_value=False,
+              prompt='Are you sure you want to delete the user?')
+def delete(ctx, user_name):
+    """
+    Deletes an user from the current organization. Also deletes all assests (e.g. vApps, vms, catalogs etc.) owned by the user. However if any of those vApps are running, the command will result in error.
+
+\b
+    Examples
```
Comment (anusuyar) [inline vcd_cli/user.py]: Do we need to put the examples in the method level? From my understanding, only the examples in the class level will be parsed and showed to the users during help option.

File: vcd_cli/user.py [hunk 3]
```diff
@@ -19,6 +21,9 @@ def user(ctx):
 \b
         vcd user create 'my user' 'my password' 'role name'
            create user in the current organization with 'my user' 'my password' and 'role name' .
+\b
+        vcd user delete 'my user'
+           deletes user with username "my user" from the current organization. Will also delete vApps owned by the user. If user has running vApps, this command will result in error.
```
Comment (anusuyar) [inline vcd_cli/user.py]: I feel its better to move the side-effects of the command as a prompt when the command is executed.

File: vcd_cli/user.py (hunk 2 above)
Comment (rocknes) [reply vcd_cli/user.py]: I tried running the following command
vcd user delete --help
and got back an empty help section. Having extra help at command level doesn't seem like a bad idea to me.

File: vcd_cli/user.py (hunk 3 above)
Comment (rocknes) [reply vcd_cli/user.py]: man/help pages often list side effects.  However I agree that we should prompt the user for confirmation along with the potential pitfall message.

File: vcd_cli/user.py (hunk 2 above)
Comment (pacogomez) [reply vcd_cli/user.py]: we are aggregating all the examples at the `<object>` subcommand level and using the default help at the `<verb>` level. I think we should keep doing that for consistency

Comment (rocknes) [reply vcd_cli/user.py]: Sure, I will make the changes.

**Pull request merged**